app/services/green/green.ts

```diff
@@ -0,0 +1,514 @@
+  }
+
+  get contextsToStream() {
+    return Object.entries(this.activeDisplayPlatforms).reduce(
+      (contextNames: TDisplayType[], [key, value]: [TDisplayType, TPlatform[]]) => {
+        if (value.length > 0) {
+          contextNames.push(key as TDisplayType);
+        }
+        return contextNames;
+      },
+      [],
+    );
+  }
+
+  get activeDisplayPlatforms() {
+    const enabledPlatforms = this.streamingService.views.enabledPlatforms;
+    return Object.entries(this.state.platformSettings).reduce(
+      (displayPlatforms: TDisplayPlatforms, [key, val]: [string, IGreenPlatformSetting]) => {
+        if (val && enabledPlatforms.includes(val.platform)) {
+          displayPlatforms[val.display].push(val.platform);
+        }
+        return displayPlatforms;
+      },
+      { horizontal: [], green: [] },
+    );
+  }
+
+  get nodeMaps() {
+    return this.state.nodeMaps;
+  }
+
+  get sceneNodeMaps() {
+    return this.state.sceneNodeMaps;
+  }
+
+  get videoSettings(): IDisplayVideoSettings {
+    return this.state.videoSettings;
+  }
+
+  get activeDisplays() {
+    return this.state.videoSettings.activeDisplays;
+  }
+
+  get defaultDisplay() {
+    const active = Object.entries(this.state.videoSettings.activeDisplays).map(([key, value]) => {
+      if (value === true) {
+        return { key };
+      }
+    });
+    return active.length > 1 ? null : active[0];
+  }
+
+  getNodeIds(displays: TDisplayType[]) {
+    return displays.reduce((ids: string[], display: TDisplayType) => {
+      const nodeMap = this.state.nodeMaps[display];
+      const aggregatedIds = Object.values(nodeMap).concat(ids);
+
+      return aggregatedIds;
+    }, []);
+  }
+
+  getPlatformDisplay(platform: TPlatform) {
+    return this.state.platformSettings[platform].display;
+  }
+
+  getPlatformContext(platform: TPlatform) {
+    const display = this.getPlatformDisplay(platform);
+    return this.videoSettingsService.contexts[display];
+  }
+
+  getDisplayNodeMap(display: TDisplayType) {
+    return this.state.nodeMaps[display];
+  }
+
+  getDisplayNodeId(defaultNodeId: string) {
+    return this.state.sceneNodeMaps[this.scenesService.views.activeSceneId][defaultNodeId];
+  }
+
+  getDisplayNodeVisibility(defaultNodeId: string, display?: TDisplayType) {
+    if (display === 'horizontal') {
+      return this.scenesService.views.getNodeVisibility(defaultNodeId);
+    } else {
+      const nodeId = this.getDisplayNodeId(defaultNodeId);
+      return this.scenesService.views.getNodeVisibility(nodeId);
+    }
+  }
+
+  getGreenNodeIds(sceneItemId: string) {
+    return [
+      this.state.nodeMaps['horizontal'][sceneItemId],
+      this.state.nodeMaps['green'][sceneItemId],
+    ];
+  }
+
+  getPlatformContextName(platform: TPlatform): TOutputOrientation {
+    return this.getPlatformDisplay(platform) === 'horizontal' ? 'landscape' : 'portrait';
+  }
+}
+
+export class GreenService extends PersistentStatefulService<IGreenServiceState> {
```

Comment:
Would be good to chat about what is being stored in persistent state here.  If it's storing stuff like IDs to scene items, then we should be persisting it in the Scene Collections system rather than with `PersistentStatefulService`

Reply:
Sounds good